ironfish/src/blockchain/blockchain.ts

```diff
@@ -985,22 +976,19 @@ export class Blockchain {
       }
 
       const blockNotes = []
-      const blockNullifiers = []
       for (const transaction of transactions) {
         for (const note of transaction.notes()) {
           blockNotes.push(note)
         }
-        for (const spend of transaction.spends()) {
-          blockNullifiers.push(spend.nullifier)
-        }
       }
 
+      const originalNullifierSize = await this.nullifiers.size(tx)
       await this.notes.addBatch(blockNotes, tx)
-      await this.nullifiers.addBatch(blockNullifiers, tx)
 
       const noteCommitment = await this.notes.rootHash(tx)
       const noteSize = await this.notes.size(tx)
-      const nullifierSize = await this.nullifiers.size(tx)
+      const addedNullifiers = transactions.reduce((count, t) => count + t.spendsLength(), 0)
+      const nullifierSize = originalNullifierSize + addedNullifiers
```

Comment:
Ok, so we're keeping this for an integrity check?

Reply:
Keeping annotated sizes on block headers for the use case I mentioned previously where we want to check if a nullifier exists in a certain confirmation range. We can get the size of the nullifier set on a specific block and check if the nullifier existed before that block

Reply:
It's hard to know how we'll want to accomplish that. Maybe we just build an in memory cache instead of using the DB here? It's unclear because it's not being used yet.

Reply:
Yea I think having an in-mem cache does make sense for this use case. I'll remove the position from this set